### PR TITLE
Socket connection/disconnection memory leak issue resolved

### PR DESCRIPTION
**Pull request by wslee94**
I tried pop3 client connection/disconnection using tls option in loop.
I confirmed through memory snapshot that th tls socket is gradually increasing and occupying memory.
So I added a code that completely disconnects the tls socket.

### CODE REVIEW

File: lib/yapople.js [hunk 1]
```diff
@@ -100,9 +100,7 @@ function onData(data) {
         , sData = data.toString()
     ;
     if (typeof this.flow === 'undefined') { // if we have first data chunk from server
-        if (sData.substr(0, 4) === '-ERR') {
-            err = new Error(sData.substring(5, sData.indexOf('\r\n')));
-        } else if (sData.substr(0, 3) === '+OK') {
```
Comment (agsh) [inline lib/yapople.js:100]: Why did you remove the error handling?

File: lib/yapople.js [hunk 2]
```diff
@@ -139,8 +137,9 @@ function onData(data) {
             this._socket.once('end', function() {
                 this.connected = false;
                 if (this._command.callback) {
-                    this._command.callback.call(this, null);
+                  this._command.callback.call(this, null);
                 }
+                this._socket.destroy(); // socket connection close
```
Comment (agsh) [inline lib/yapople.js:137]: Cool :+1:

File: package.json [hunk 3]
```diff
@@ -1,7 +1,7 @@
 {
-  "name": "yapople",
```
Comment (agsh) [inline package.json:2]: Just remove `package.json` from commit, it breaks package name and version.
Or, if you want to mention you as a contributor, you can do it like this: https://nodejs.dev/learn/the-package-json-guide#contributors
And don't forget about these gyus :smile: https://github.com/agsh/yapople/graphs/contributors

File: lib/yapople.js (hunk 1 above)
Comment (wslee94) [reply lib/yapople.js:100]: Most of the time, there was no problem.
In my case, an Unhandled Promise Rejection error occurred on line 104 due to different data formats from the mail server I access sometimes.

Therefore, I changed the code as below.

**if (sData.substr(0, 3) === '+OK') {
  succ = sData.substring(4, sData.indexOf('\r\n'));
} else {
  err = new Error(sData);
}**

File: lib/yapople.js (hunk 2 above)
Comment (wslee94) [reply lib/yapople.js:137]: Thank you.

File: package.json (hunk 3 above)
Comment (wslee94) [reply package.json:2]: I modified package.json.

**Comment by agsh** [comment]
@wslee94 Done and published in the npm

**Comment by wslee94** [comment]
Hello.

You have to modify the package.json. url in repository property is entered incorrectly, so you have to correct it to url and publish it.

image

**Comment by agsh** [comment]
@wslee94  Oh! Missed that, fixed it, thanks!
